pytorch_lightning/compressor/intel/neural_compressor/quantization.py

```diff
@@ -0,0 +1,119 @@
+from enum import Enum
+from torch.utils.data import DataLoader
+from pytorch_lightning.core.lightning import LightningModule
+from typing import Optional
+import pytorch_lightning as pl
+
+
+class QuantizationMode(Enum):
+    PTQ_DYNAMIC = "post_training_dynamic_quant"
+    PTQ_STATIC = "post_training_static_quant"
+    QAT = "quant_aware_training"
+
+
+class IncQuantization:
+    def __init__(
+            self,
+            config_path_or_obj: str,
+            model: LightningModule,
+            calib_dataloader: Optional[DataLoader] = None,
+            metric_name: str = None,
+            trainer=None
+    ):
+        """
+        Args:
+            config_path_or_obj (:obj:`str` ):
+                Path to the YAML configuration file used to control the tuning behavior.
+            model (:obj:`LightningModule`):
+                FP32 model specified for low precision tuning.
+            calib_dataloader (:obj:`DataLoader`, `optional`):
+                DataLoader for calibration.
+            metric_name(:obj:`str`):
+                metric name to symbolic trace to get FX GraphModule
+            trainer_args:
+                arguments for PyTorch Lightning trainer
+        Returns:
+            quantizer: IncQuantization object.
+        """
+        from neural_compressor.conf.config import Quantization_Conf
```

Comment:
This should fail with a good error message if neural_compressor isn't installed.

Reply:
done